tests/test_search.py

```diff
@@ -176,7 +177,7 @@ def test_is_pattern(value, expected):
             ],
         ),
         (
-            {"A": [re.compile("^a.*a$", flags=re.IGNORECASE)]},
+            {"A": ["(?i)^a.*a$"]},
```

Comment:
Polars string matching has first class support for regexes. Setting case insensitive search in polars is as straightforward as adding a `(?i)` group.

There doesn't seem to be any documentation indicating that we expose compiled regex's to the user as a way to interact with the catalog, so I think it's safe to drop the use of `re.compile` here.